### PR TITLE
Backend: Add dungeon phases

### PR DESCRIPTION
**Pull request by martimavocado**
## What
Adds dungeon phase detection

## Changelog Technical Details
+ Added dungeon phase detection. - martimavocado

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonAPI.kt [hunk 1]
```diff
@@ -343,4 +347,97 @@ object DungeonAPI {
             fun getByInventoryName(inventory: String) = entries.firstOrNull { it.inventory == inventory }
         }
     }
+
+    enum class DungeonPhase {
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonAPI.kt]: you can embed the patterns directly into the enum. Look at Skyblock Guide Highlight as an example of how (Not a good example but the first that came to mind)

**Comment by catgirlseraid** [comment]
change phasePatternGroup to patternGroup 🥺

**Comment by martimavocado** [comment]
> change phasePatternGroup to patternGroup 🥺

there's already a `patternGroup` though do i just make `phasePatternGroup` a subgroup inside `patternGroup`? or i could just add `boss.message` to every key in the new patterns i added and ditch `phasePatternGroup`

**Review by hannibal002 — CHANGES_REQUESTED** [review]
please add regex tests to all patterns

**Comment by martimavocado** [comment]
> please add regex tests to all patterns

is there a point in doing this for most of these patterns? only one isn't checking for an exact string. i don't mind doing this, just curious as to why

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Please move the code from DungeonAPI into a new class DungeonBossAPI or similar, and change the name of the event and the enum to indlicate that this is not the dungeon phase but the boss phase.

**Review by hannibal002 — CHANGES_REQUESTED** [review]
the phase doesnt reset to null after leaving dungeon

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**